### PR TITLE
Octane login and fields integration

### PR DESCRIPTION
**Pull request by Maheshkale447**
Added functions to get Octane projects, domains and fields

### CODE REVIEW

File: Ginger/GingerCore/ALM/OctaneCore.cs [hunk 1]
```diff
@@ -15,57 +15,78 @@
...
-                }
+                        return octaneRepository.IsLoginValid(
+                            new LoginDTO()
+                            {
+                                User = ALMCore.DefaultAlmConfig.ALMUserName,
+                                Password = ALMCore.DefaultAlmConfig.ALMPassword,
+                                Server = ALMCore.DefaultAlmConfig.ALMServerURL
+                            });
+                    }).Result;
+                //mOctaneRestConnector = new RestConnector();
```
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/OctaneCore.cs:63]: Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6235433)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 5
           

Complexity increasing per file
==============================
- Ginger/GingerCore/ALM/OctaneCore.cs  11
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6235433&bid=20317248)

File: Ginger/GingerCore/ALM/OctaneCore.cs [hunk 2]
```diff
@@ -105,27 +123,132 @@ public override bool ExportExecutionDetailsToALM(BusinessFlow bizFlow, ref strin
...
+                    {
+                        itemfield.ToUpdate = true;
+                    }
+                    itemfield.ItemType = entityType;
+                    itemfield.Type = field.FieldType;
+
+                    if (listnodes != null && listnodes.ContainsKey(field.Name) && listnodes[field.Name].Any())
+                        itemfield.PossibleValues = new ObservableList<string>(listnodes[field.Name]);
+
+                    else if (listnodes != null && listnodes.ContainsKey(entityType.ToLower() + "_" +field.Name) && listnodes[entityType.ToLower() + "_" + field.Name].Any())
```
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/OctaneCore.cs:236]: Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6235433)

File: Ginger/GingerCore/ALM/OctaneCore.cs [hunk 3]
```diff
@@ -15,57 +15,78 @@
...
         {
-            throw new NotImplementedException();
+            return this.ConnectALMServer();
         }
 
         public override bool ConnectALMServer()
         {
             try
             {
+                if (octaneRepository == null)
```
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/OctaneCore.cs:50]: Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6235433)

File: Ginger/GingerCore/ALM/OctaneCore.cs [hunk 4]
```diff
@@ -105,27 +123,132 @@ public override bool ExportExecutionDetailsToALM(BusinessFlow bizFlow, ref strin
...
+                    itemfield.ItemType = entityType;
+                    itemfield.Type = field.FieldType;
+
+                    if (listnodes != null && listnodes.ContainsKey(field.Name) && listnodes[field.Name].Any())
+                        itemfield.PossibleValues = new ObservableList<string>(listnodes[field.Name]);
+
+                    else if (listnodes != null && listnodes.ContainsKey(entityType.ToLower() + "_" +field.Name) && listnodes[entityType.ToLower() + "_" + field.Name].Any())
+                        itemfield.PossibleValues = new ObservableList<string>(listnodes[entityType.ToLower() + "_" + field.Name]);
+
+                    else if (phases != null && phases.ContainsKey(field.Name) && phases[field.Name].Any())
```
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/OctaneCore.cs:239]: Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6235433)

File: Ginger/GingerCore/ALM/OctaneCore.cs [hunk 5]
```diff
@@ -105,27 +123,132 @@ public override bool ExportExecutionDetailsToALM(BusinessFlow bizFlow, ref strin
...
+                    itemfield.SystemFieled = !field.IsUserField;
+                    
+                    if (itemfield.Mandatory)
+                    {
+                        itemfield.ToUpdate = true;
+                    }
+                    itemfield.ItemType = entityType;
+                    itemfield.Type = field.FieldType;
+
+                    if (listnodes != null && listnodes.ContainsKey(field.Name) && listnodes[field.Name].Any())
```
Comment (ambarkhan) [inline Ginger/GingerCore/ALM/OctaneCore.cs:233]: Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6235433)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- Ginger/GingerCore/ALM/OctaneCore.cs  11
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6235433&bid=20317248)

**Review by JinendraGhodke — APPROVED** [review]

**Pull request merged**